nf_core/modules/lint/main_nf.py

```diff
@@ -241,11 +241,11 @@ def check_process_section(self, lines):
         if l.startswith("https://containers") or l.startswith("https://depot"):
             # e.g. "https://containers.biocontainers.pro/s3/SingImgsRepo/biocontainers/v1.2.0_cv1/biocontainers_v1.2.0_cv1.img' :" -> v1.2.0_cv1
             # e.g. "https://depot.galaxyproject.org/singularity/fastqc:0.11.9--0' :" -> 0.11.9--0
-            singularity_tag = re.search("(?:\/)?(?:biocontainers_)?(?::)?([A-Za-z\d\-_\.]+)(?:\.img)?['\"]", l).group(1)
+            singularity_tag = re.search(r"(?:/)?(?:biocontainers_)?(?::)?([A-Za-z\d\-_.]+)(?:\.img)?['\"]", l).group(1)
```

Comment:
Hi, I was checking this again, I'm fine with removing the backslash as they are not needed, but still captures .img, which I thought I had solved when I coded it! 
Anyway, playing around, I think that making the `+` non-greedy with `+?` makes the trick :) 

```suggestion
            singularity_tag = re.search(r"(?:/)?(?:biocontainers_)?(?::)?([A-Za-z\d\-_.]+?)(?:\.img)?['\"]", l).group(1)
```

Reply:
Oh, sorry for the confusion. I just reverted the two commits with the regex fix this morning to have this PR handle string representation and formatting only. The regex-matching-fix is exactly as you figured it out in #1598

Reply:
I also updated this PR opening post accordingly and removed the regex-fix. Sorry again for the confusion. I just thought that the regex fix might be merged faster if it is alone in case there is some opposition to standardizing the string formatting.

Reply:
Oh I see, haven't seen the other PR sorry, all done then, thanks!

Reply:
@mirpedrol #1595 even has links showing the regex behaviour pre and post #1598